pkg/component/extensions/operatingsystemconfig/nodeinit/templates/scripts/init.tpl.sh

```diff
@@ -11,20 +11,22 @@ unmount() {
 }
 trap unmount EXIT
 
-echo "> Pull gardener-node-agent image and mount it to the temporary directory"
+echo "> Pull {{ .binaryName }} image and mount it to the temporary directory"
 ctr images pull --hosts-dir "/etc/containerd/certs.d" "{{ .image }}"
 ctr images mount "{{ .image }}" "$tmp_dir"
 
-echo "> Copy gardener-node-agent binary to host ({{ .binaryDirectory }}) and make it executable"
+echo "> Copy {{ .binaryName }} binary to host ({{ .binaryDirectory }}) and make it executable"
 mkdir -p "{{ .binaryDirectory }}"
 
 {{- /*
-Fall back to /ko-app/gardener-node-agent if /gardener-node-agent doesn't exist in image to support images built with ko.
+Fall back to /ko-app/<binary-name> if /<binary-name> doesn't exist in image to support images built with ko.
 TODO(timebertt): remove this fallback once https://github.com/ko-build/ko/pull/1403 has been released and is used to
  build images in the skaffold-based setup (add a breaking release note!).
 */}}
-cp -f "$tmp_dir/gardener-node-agent" "{{ .binaryDirectory }}" || cp -f "$tmp_dir/ko-app/gardener-node-agent" "{{ .binaryDirectory }}"
-chmod +x "{{ .binaryDirectory }}/gardener-node-agent"
+cp -f "$tmp_dir/{{ .binaryName }}" "{{ .binaryDirectory }}" || cp -f "$tmp_dir/ko-app/{{ .binaryName }}" "{{ .binaryDirectory }}"
+chmod +x "{{ .binaryDirectory }}/{{ .binaryName }}"
+{{- if eq .binaryName "gardener-node-agent" }}
```

Comment:
```suggestion

{{- if eq .binaryName "gardener-node-agent" }}
```

Nit for readability 😇